libs/shared/src/services/core/analytics.service.ts

```diff
@@ -35,4 +35,20 @@ export class AnalyticsService {
       }
     });
   }
+
+  // Method to track when users play a video
+  public trackVideoPlay(videoId: string) {
+    this.firebaseAnalytics.logEvent({
+      name: 'video_play',
+      params: { video_id: videoId, app_version: APP_VERSION },
+    });
+  }
+
+  // Method to track when users opens resource file
+  public trackResourceOpen() {
+    this.firebaseAnalytics.logEvent({
+      name: 'open_resource_file',
+      params: { app_version: APP_VERSION },
```

Comment:
nit(non-blocking): same as above, although would also be nice to have the `resource_id` tracked as part of the the payload sent to analytics so that we can track research questions such as which resources are accessed most frequently

Reply:
Will work on that.